cypress/mocks/handlers.ts

```diff
@@ -0,0 +1,128 @@
+    if (!session) {
+      return new HttpResponse('Session not found', { status: 404 });
+    }
+    return HttpResponse.json(session);
+  }),
+
+  // Mock for creating a session
+  http.post(`${API_BASE_URL}/session`, async ({ request }) => {
+    const payload = (await request.json()) as Session;
+    const lastSessionId = Math.max(...Array.from(sessions.keys()), 0);
+
+    const newSession = {
+      ...payload,
+      id: lastSessionId + 1,
+      meta_data: { ...payload.meta_data, status: STATUS.SUCCESS },
+    };
+    sessions.set(newSession.id, newSession);
+    return HttpResponse.json(newSession);
+  }),
+
+  // Mock for patching a session
+  http.patch(`${API_BASE_URL}/session/:id`, async ({ params, request }) => {
+    const { id } = params;
+    const payload = (await request.json()) as Session;
+    const sessionId = parseInt(id as string);
+
+    if (sessions.has(sessionId)) {
+      const existingSession = sessions.get(sessionId);
+      sessions.set(sessionId, { ...existingSession, ...payload });
+      mock_sns({ oldSession: sessions.get(sessionId) as Session });
+      return HttpResponse.json(sessions.get(sessionId));
+    }
+
+    return new HttpResponse('Session not found', { status: 404 });
+  }),
+
+  // Mock for fetching auth groups
+  http.get(`${API_BASE_URL}/auth-group`, async ({ request }) => {
+    const authGroups = await fetch(bypass(request)).then((response) => response.json());
+    return HttpResponse.json(authGroups);
+  }),
+
+  // Mock for fetching batches
+  http.get(`${API_BASE_URL}/batch`, async ({ request }) => {
+    const batches = await fetch(bypass(request)).then((response) => response.json());
+    return HttpResponse.json(batches);
+  }),
+
+  // Mock for fetching form schemas
+  http.get(`${API_BASE_URL}/form-schema`, async ({ request }) => {
+    const formSchemas = await fetch(bypass(request)).then((response) => response.json());
+    return HttpResponse.json(formSchemas);
+  }),
+];
+
+async function mock_sns({ oldSession }: { oldSession: Session }) {
+  let payload: Session = {};
+  if (oldSession.platform === Platform.Quiz) {
+    const { name, activeDays, endDate, startDate } = MockQuizData.edit;
+    payload = {
+      name: name,
+      end_time: utcToISTDate(endDate.toUTCString()),
+      start_time: utcToISTDate(startDate.toUTCString()),
+      repeat_schedule: {
+        type: 'weekly',
+        params: activeDays,
+      },
+    };
+  } else {
+    const { name, activeDays, endDate, startDate } = MockLiveData.edit;
+    payload = {
+      name: name,
+      end_time: utcToISTDate(endDate.toUTCString()),
+      start_time: utcToISTDate(startDate.toUTCString()),
+      repeat_schedule: {
+        type: 'weekly',
+        params: activeDays,
+      },
+    };
+  }
+  const newSession = {
+    ...oldSession,
+    ...payload,
+    meta_data: {
+      ...(oldSession?.meta_data ?? {}),
+      ...(payload?.meta_data ?? {}),
+      status: STATUS.SUCCESS,
+    },
+  };
+
+  await delay(20000);
```

Comment:
whats this delay for

Reply:
Just Mocking the Lambda as mostly it takes 20s to complete the patch. 
and using this I'm testing the polling in the frontend.

Reply:
20s is a lot i think. it will make the testing slow. i dont know if this 20s is real time or assumed (compressed) cypress time

Reply:
Changes it to 4sec.